sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultContainerSpec.java

```diff
@@ -5,27 +5,35 @@
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 
-import javax.validation.constraints.NotNull;
+import javax.validation.Valid;
 import javax.validation.constraints.Size;
+import java.util.Optional;
 
 /**
  * Default implementation of {@link ContainerSpec}.
  */
 public class DefaultContainerSpec implements ContainerSpec {
-    @NotNull
     @Size(min = 1)
     private String imageName;
+    @Valid
+    private RLimitSpec rLimitSpec;
 
     @JsonCreator
     public DefaultContainerSpec(
-            @JsonProperty("image-name")
-            String imageName) {
+            @JsonProperty("image-name") String imageName,
+            @JsonProperty("rlimit_spec") RLimitSpec rLimitSpec) {
         this.imageName = imageName;
+        this.rLimitSpec = rLimitSpec;
     }
 
     @Override
-    public String getImageName() {
-        return imageName;
+    public Optional<String> getImageName() {
+        return Optional.ofNullable(imageName);
+    }
+
+    @Override
+    public Optional<RLimitSpec> getRLimitSpec() {
```

Comment:
Maybe just a `Collection<RLimit>` named `rlimits` here? It looks like the current schema would work out as:

```
"container" : {
  "rlimit-spec": {
    "rlimits": [
      ...
    ]
  }
}
```

When it could just be:

```
"container" : {
  "rlimits": [
    ...
  ]
}
```